### PR TITLE
feat: add as_secret feature

### PR DESCRIPTION
**Pull request by JeyJeyGao**
feature:
- added `as_secret` feature to get the certificate chain with GetSecret permission

Tested with Notation CLI:
- PEM certificate chain with GetSecret permission
- PKCS12 certificate chain with GetSecret Permission

Resolve part of #90 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs [hunk 1]
```diff
@@ -22,39 +22,51 @@ public async Task<object> RunAsync(string inputJson)
...
-                certificateChain = CertificateChain.Build(leafCert, CertificateBundle.Create(caCertsPath));
+                // Obtain the certificate bundle from file 
+                // (including the intermediate and root certificates).
+                certBundle = CertificateBundle.Create(input.PluginConfig["ca_certs"]);
+
+                // obtain the leaf certificate from Azure Key Vault
+                leafCert = await akvClient.GetCertificateAsync();
             }
-            else if (input.PluginConfig?.ContainsKey("as_secret") == true)
+            else if (input.PluginConfig?.GetValueOrDefault("as_secret")?.Equals("true", StringComparison.OrdinalIgnoreCase) == true)
```
Comment (shizhMSFT) [inline Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs:37]: I'm not sure if `TryGetValue()` can also apply here or not.

File: Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs [hunk 2]
```diff
@@ -22,39 +22,51 @@ public async Task<object> RunAsync(string inputJson)
...
             {
-                // validate the self-signed leaf certificate
-                certificateChain = CertificateChain.Build(leafCert, new X509Certificate2Collection());
+                // only have the leaf certificate
+                certBundle = new X509Certificate2Collection();
+                leafCert = await akvClient.GetCertificateAsync();
             }
 
+            // Extract KeySpec from the certificate
+            var keySpec = leafCert.KeySpec();
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs:60]: In fact, line 63 and line 68 are different values.

> line 63: var signature = await akvClient.SignAsync(keySpec.To**Signature**Algorithm(), input.Payload);

> line 68:                 signingAlgorithm: keySpec.To**Signing**Algorithm(),

I renamed the keySpec.To**Signature**Algorithm() to be keySpec.ToKeyVaultSignatureAlgorithm() to make it clear.

File: Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs [hunk 3]
```diff
@@ -22,39 +22,51 @@ public async Task<object> RunAsync(string inputJson)
...
-
-            // Sign
-            var signature = await akvClient.SignAsync(signatureAlgorithm, input.Payload);
-
-            // Build the certificate chain
-            List<byte[]> certificateChain = new List<byte[]>();
+            // Obtain the certificate chain
+            X509Certificate2Collection certBundle;
+            X509Certificate2 leafCert;
             if (input.PluginConfig?.ContainsKey("ca_certs") == true)
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs]: Updated.

File: Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs (hunk 1 above)
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs:60]: LoL. I was confused. Thanks for the update.

**Pull request merged**